cmd/tether/attach.go

```diff
@@ -179,131 +168,147 @@ func (t *attachServerSSH) stop() {
 func (t *attachServerSSH) run() error {
 	defer trace.End(trace.Begin("main attach server loop"))
 
-	var sConn *ssh.ServerConn
+	var serverConn *ssh.ServerConn
 	var chans <-chan ssh.NewChannel
 	var reqs <-chan *ssh.Request
 	var err error
 
-	// keep waiting for the connection to establish
-	for t.enabled && sConn == nil {
-		conn := t.conn
```

Comment:
was specifically using a copy of conn as it was getting nil'd elsewhere (presumably by stop) - doesn't look like we've got a lock here, so that possibility still exists. We cannot just hold the lock here while we access the conn because we cannot yield it during all the blocking operations, hence the copy.